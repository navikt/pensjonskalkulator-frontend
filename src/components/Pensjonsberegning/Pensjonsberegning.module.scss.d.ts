import globalClassNames from '../../style.d'
declare const classNames: typeof globalClassNames & {
  readonly section: 'section'
  readonly ingress: 'ingress'
  readonly loader: 'loader'
  readonly chipsWrapper: 'chipsWrapper'
  readonly chipsWrapper__hasGap: 'chipsWrapper__hasGap'
  readonly 'navds-chips--icon-left': 'navds-chips--icon-left'
  readonly 'navds-chips__toggle-icon': 'navds-chips__toggle-icon'
  readonly visFlereAldere: 'visFlereAldere'
  readonly visTabell: 'visTabell'
}
export = classNames
