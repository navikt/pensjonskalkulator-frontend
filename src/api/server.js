// const express = require('express')
// const liveness = require('../src/__mocks__/data/liveness.json')
// const pensjonsberegning = require('../src/__mocks__/data/pensjonsberegning.json')

// const app = express()
// const port = 8088

// app.get('/api/status', (request, response) => {
//   response.send(liveness)
// })

// app.get('/api/pensjonsberegning', async (request, response) => {
//   response.send(pensjonsberegning)
// })

// app.listen(process.env.PORT || port, () => {
//   console.info('proxy server is running on port', process.env.PORT || port)
// })

// import { rest } from 'msw'
// import { setupServer } from 'msw/node'

import { setupServer } from 'msw/node'
import { handlers } from './handlers.js'

export const server = setupServer(...handlers)
server.listen({ onUnhandledRequest: 'error' })
