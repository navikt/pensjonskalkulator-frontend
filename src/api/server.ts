import { setupServer } from 'msw/node'
import { getHandlers } from './handlers.js'
import { rest } from 'msw'

export const API_TARGET = 'http://localhost:8088'
export const API_PATH = '/pensjon/kalkulator/api'

const handlers = getHandlers(`${API_TARGET}${API_PATH}`)
export const server = setupServer(...handlers)

server.listen({ onUnhandledRequest: 'error' })

type MockResponseOptions = {
  status?: number
  json?: any
}

export const mockResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 200,
    json: 'OK',
  }
) => {
  server.use(
    rest.get(`${API_TARGET}${API_PATH}${path}`, (req, res, ctx) => {
      return res(ctx.status(options.status ?? 200), ctx.json(options.json))
    })
  )
}

export const mockErrorResponse = (
  path: string,
  options: MockResponseOptions = {
    status: 500,
    json: "Beep boop I'm an error!",
  }
) => {
  mockResponse(path, {
    ...options,
    status: options.status ?? 500,
  })
}
