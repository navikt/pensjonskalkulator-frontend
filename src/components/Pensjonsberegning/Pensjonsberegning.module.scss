@import '../../scss/variables';

.section {
  gap: var(--a-spacing-4);
}

.ingress {
  max-width: $paragraph-max-width;
}

.loader {
  margin: var(--a-spacing-6) auto;
}

.chipsWrapper {
  /* double-specificity n√∏dvendig pga Aksel overskrivning */
  &.chipsWrapper__hasGap {
    gap: var(--a-spacing-3);
    justify-content: space-between;

    &::after {
      content: '';
      flex: auto;
    }
  }

  :global {
    .navds-chips--icon-left {
      padding-left: var(--a-spacing-3);
    }

    .navds-chips__toggle-icon {
      display: none;
    }
  }
}

.visFlereAldere {
  margin-right: auto;
  margin-bottom: var(--a-spacing-3);
}

.visTabell {
  margin: 0 auto;
}
