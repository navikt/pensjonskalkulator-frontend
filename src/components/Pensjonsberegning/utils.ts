export const generateAlderArray = (
  startAlder: number,
  endAlder: number,
  formatertTidligsteMuligeUttaksalder: string
) => {
  const alderArray: string[] = []
  for (let i = startAlder; i <= endAlder; i++) {
    if (i === startAlder) {
      alderArray.push(formatertTidligsteMuligeUttaksalder)
    } else {
      alderArray.push(`${i.toString()} 책r`)
    }
  }
  return alderArray
}

export const formatUttaksalder = (
  { aar, maaned }: Uttaksalder,
  options: { compact: boolean } = { compact: false }
): string => {
  return maaned !== 0
    ? `${aar} 책r og ${maaned} ${options.compact ? 'md.' : 'm책neder'}`
    : `${aar} 책r`
}
