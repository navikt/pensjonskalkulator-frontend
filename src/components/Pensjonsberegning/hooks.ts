import { useMemo } from 'react'

import { generateAlderArray, formatUttaksalder } from './utils'

export const useAlderChips = (data?: Uttaksalder, maksalder = 77): string[] =>
  useMemo(
    () =>
      data?.aar
        ? generateAlderArray(
            data.aar,
            maksalder,
            formatUttaksalder(data, { compact: true })
          )
        : [],
    [data]
  )
