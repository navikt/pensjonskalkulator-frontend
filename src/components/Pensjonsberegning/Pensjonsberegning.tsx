import React, { useMemo, useState } from 'react'

import { ChevronDownIcon, ChevronUpIcon } from '@navikt/aksel-icons'
import {
  Alert,
  Button,
  Chips,
  Heading,
  Ingress,
  Loader,
  ReadMore,
} from '@navikt/ds-react'
import clsx from 'clsx'

import { Pensjonssimulering } from '../Pensjonssimulering'
import { Card } from '@/components/Card'
import { Forbehold } from '@/components/Forbehold'
import { Grunnlag } from '@/components/Grunnlag'
import { useGetTidligsteMuligeUttaksalderQuery } from '@/state/api/apiSlice'

import { useAlderChips } from './hooks'
import { formatUttaksalder } from './utils'

import styles from './Pensjonsberegning.module.scss'

export function Pensjonsberegning() {
  const {
    data: tidligstMuligUttak,
    isLoading,
    isError,
    isSuccess,
  } = useGetTidligsteMuligeUttaksalderQuery()
  const DEFAULT_ANTALL_VISIBLE_ALDERCHIPS = 9
  const VIS_FLERE__ALDERE_LABEL_CLOSE = 'Vis flere aldere'
  const VIS_FLERE__ALDERE_LABEL_OPEN = 'Vis færre aldere'

  const [isFlereAldereOpen, setIsFlereAldereOpen] = useState<boolean>(false)

  const [valgtUttaksalder, setValgtUttaksalder] = useState<string | undefined>()

  const alderChips = useAlderChips(tidligstMuligUttak)
  const formatertUttaksalder = useMemo(() => {
    return isSuccess ? formatUttaksalder(tidligstMuligUttak) : null
  }, [isSuccess, tidligstMuligUttak])

  if (isLoading) {
    return (
      <Loader
        className={styles.loader}
        data-testid="loader"
        size="3xlarge"
        title="Henter tidligste mulige uttaksalder"
      />
    )
  }

  if (isError || !isSuccess) {
    return (
      <Alert variant="error">
        <Heading spacing size="small" level="2">
          Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen
          senere.
        </Heading>
      </Alert>
    )
  }

  return (
    <>
      <Card data-testid="tidligst-mulig-uttak">
        <Ingress>
          Du kan tidligst ta ut alderspensjon når du er {formatertUttaksalder}.
          Hvis du går av senere, får du høyere pensjon i året.
        </Ingress>
      </Card>

      <Card className={styles.section}>
        <Heading size="xsmall" level="2">
          Når vil du ta ut alderspensjon?
        </Heading>
        <Chips
          className={clsx(styles.chipsWrapper, styles.chipsWrapper__hasGap)}
        >
          {alderChips.length > 0 &&
            alderChips
              .slice(
                0,
                isFlereAldereOpen
                  ? alderChips.length
                  : DEFAULT_ANTALL_VISIBLE_ALDERCHIPS
              )
              .map((alderChip) => (
                <Chips.Toggle
                  selected={valgtUttaksalder === alderChip}
                  key={alderChip}
                  onClick={() => setValgtUttaksalder(alderChip)}
                >
                  {alderChip}
                </Chips.Toggle>
              ))}
        </Chips>
        <Button
          className={styles.visFlereAldere}
          icon={
            isFlereAldereOpen ? (
              <ChevronUpIcon aria-hidden />
            ) : (
              <ChevronDownIcon aria-hidden />
            )
          }
          iconPosition="left"
          size={'xsmall'}
          variant="tertiary"
          onClick={() => {
            setIsFlereAldereOpen(!isFlereAldereOpen)
          }}
        >
          {isFlereAldereOpen
            ? VIS_FLERE__ALDERE_LABEL_OPEN
            : VIS_FLERE__ALDERE_LABEL_CLOSE}
        </Button>

        {valgtUttaksalder && (
          <>
            <Pensjonssimulering uttaksalder={parseInt(valgtUttaksalder, 10)} />
            <ReadMore
              header={'Vis tabell'}
              className={styles.visTabell}
              // icon={<ChevronDownIcon aria-hidden />}
              // iconPosition="left"
              // size={'medium'}
              // variant="tertiary"
            >
              <p>{'//TODO'}</p>
            </ReadMore>
          </>
        )}
      </Card>

      {valgtUttaksalder && (
        <>
          <Grunnlag tidligstMuligUttak={tidligstMuligUttak} />
          <Forbehold />
        </>
      )}
    </>
  )
}
