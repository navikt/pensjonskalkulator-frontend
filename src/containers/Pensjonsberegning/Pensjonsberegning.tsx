import React, { useState } from 'react'

import { Alert, Heading } from '@navikt/ds-react'

import { Card } from '@/components/Card'
import { Forbehold } from '@/components/Forbehold'
import { Grunnlag } from '@/components/Grunnlag'
import { Loader } from '@/components/Loader'
import { Pensjonssimulering } from '@/components/Pensjonssimulering'
import { TidligstMuligUttaksalder } from '@/components/TidligstMuligUttaksalder'
import { TilbakeEllerAvslutt } from '@/components/TilbakeEllerAvslutt'
import { VelgUttaksalder } from '@/components/VelgUttaksalder'
import { useTidligsteUttaksalder } from '@/routes/hooks'

export function Pensjonsberegning() {
  const {
    data: tidligstMuligUttak,
    isLoading,
    isError,
    isSuccess,
  } = useTidligsteUttaksalder()

  const [valgtUttaksalder, setValgtUttaksalder] = useState<string | undefined>()

  if (isLoading) {
    return (
      <Loader
        data-testid="loader"
        size="3xlarge"
        title="Henter tidligste mulige uttaksalder"
      />
    )
  }

  if (isError || !isSuccess) {
    return (
      <Alert variant="error">
        <Heading spacing size="small" level="2">
          Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen
          senere.
        </Heading>
      </Alert>
    )
  }

  return (
    <>
      <TidligstMuligUttaksalder uttaksalder={tidligstMuligUttak} />

      <Card>
        <VelgUttaksalder
          tidligstMuligUttak={tidligstMuligUttak}
          valgtUttaksalder={valgtUttaksalder}
          setValgtUttaksalder={setValgtUttaksalder}
        />

        {valgtUttaksalder && (
          <Pensjonssimulering uttaksalder={parseInt(valgtUttaksalder, 10)} />
        )}
      </Card>

      {valgtUttaksalder && (
        <>
          <Grunnlag tidligstMuligUttak={tidligstMuligUttak} />
          <Forbehold />
        </>
      )}

      <TilbakeEllerAvslutt />
    </>
  )
}
