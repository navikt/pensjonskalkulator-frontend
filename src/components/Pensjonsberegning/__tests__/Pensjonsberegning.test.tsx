import { describe, expect, it, vi } from 'vitest'

import { Pensjonsberegning } from '../Pensjonsberegning'
import { mockErrorResponse, mockResponse } from '@/mocks/server'
import {
  fireEvent,
  render,
  screen,
  swallowErrorsAsync,
  waitFor,
} from '@/test-utils'

describe('Pensjonsberegning', () => {
  it('viser loading og deretter riktig header, tekst og knapper', async () => {
    const result = render(<Pensjonsberegning />)

    expect(screen.getByTestId('loader')).toBeVisible()

    await waitFor(() => {
      expect(screen.queryByTestId('loader')).not.toBeInTheDocument()
    })

    expect(screen.getByTestId('tidligst-mulig-uttak')).toBeVisible()
    expect(screen.getByRole('heading', { level: 2 })).toBeVisible()
    expect(screen.getAllByRole('button')).toHaveLength(10)

    expect(result.asFragment()).toMatchSnapshot()
  })

  it('viser riktig label og antall knapper når brukeren ønsker å se flere aldere', async () => {
    const result = render(<Pensjonsberegning />)
    await waitFor(() => {
      expect(screen.getAllByRole('button')).toHaveLength(10)
    })
    fireEvent.click(screen.getByText('Vis flere aldere'))

    await waitFor(() => {
      expect(screen.getAllByRole('button')).toHaveLength(15)
      expect(screen.getByText('Vis færre aldere')).toBeInTheDocument()
    })

    expect(result.asFragment()).toMatchSnapshot()
  })

  it('viser feilmelding om henting av pensjonberegning feiler', async () => {
    mockErrorResponse('/tidligste-uttaksalder')

    const result = render(<Pensjonsberegning />)

    await waitFor(() => {
      expect(
        screen.getByText(
          'Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.'
        )
      ).toBeVisible()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('viser feilmelding om pensjonsberegning er på ugyldig format', async () => {
    const invalidData = {
      aar: 67,
      maaned: null,
    } as unknown as Uttaksalder
    mockResponse('/tidligste-uttaksalder', { json: [invalidData] })

    render(<Pensjonsberegning />)

    await swallowErrorsAsync(async () => {
      await waitFor(() => {
        expect(
          screen.getByText(
            'Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.'
          )
        ).toBeVisible()
      })
    })
  })

  it('oppdaterer valgt knapp og tegner graph når brukeren klikker på en knapp', async () => {
    const { container } = render(<Pensjonsberegning />)

    await waitFor(async () => {
      fireEvent.click(screen.getByText('Vis flere aldere'))
      fireEvent.click(screen.getByText('72 år'))

      expect(screen.getByRole('button', { pressed: true })).toHaveTextContent(
        '72 år'
      )
      vi.useFakeTimers()
      vi.advanceTimersByTime(250)
      expect(
        container.getElementsByClassName('highcharts-container').length
      ).toBe(1)

      vi.useRealTimers()
    })
  })
})
