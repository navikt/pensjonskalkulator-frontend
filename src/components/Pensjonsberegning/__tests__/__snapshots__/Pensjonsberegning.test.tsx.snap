// Vitest Snapshot v1

exports[`Pensjonsberegning > viser feilmelding om henting av pensjonberegning feiler 1`] = `
<DocumentFragment>
  <div
    class="navds-alert navds-alert--error navds-alert--medium"
  >
    <svg
      aria-labelledby="title-r6"
      class="navds-alert__icon"
      fill="none"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title-r6"
      >
        Feil
      </title>
      <path
        clip-rule="evenodd"
        d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm3.571 7L17 8.429 13.428 12 17 15.571 15.571 17 12 13.428 8.429 17 7 15.571 10.572 12 7 8.429 8.429 7 12 10.572 15.571 7Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="navds-alert__wrapper navds-body-long"
    >
      <h2
        class="navds-heading navds-heading--small navds-typo--spacing"
      >
        Vi klarte ikke å hente din tidligste mulige uttaksalder. Prøv igjen senere.
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pensjonsberegning > viser loading og deretter riktig header, tekst og knapper 1`] = `
<DocumentFragment>
  <section
    class="_section_1aa5c0"
    data-testid="tidligst-mulig-uttak"
  >
    <p
      class="_ingress_01a375 navds-ingress"
    >
      Du kan tidligst ta ut alderspensjon når du er 62 år og 10 måneder. Hvis du går av senere, får du høyere pensjon i året.
    </p>
  </section>
  <section
    class="_section_1aa5c0 _section_01a375"
  >
    <h2
      class="navds-heading navds-heading--xsmall"
    >
      Når vil du ta ut alderspensjon?
    </h2>
    <ul
      class="navds-chips _chipsWrapper_01a375 _chipsWrapper__hasGap_01a375 navds-chips--medium navds-body-short navds-body-short--small"
    >
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            62 år og 10 md.
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            63 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            64 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            65 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            66 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            67 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            68 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            69 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            70 år
          </span>
        </button>
      </li>
    </ul>
    <button
      class="_visFlereAldere_01a375 navds-button navds-button--tertiary navds-button--xsmall"
    >
      <span
        class="navds-button__icon"
      >
        <svg
          aria-hidden="true"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        aria-live="polite"
        class="navds-label navds-label--small"
      >
        Vis flere aldere
      </span>
    </button>
  </section>
</DocumentFragment>
`;

exports[`Pensjonsberegning > viser riktig label og antall knapper når brukeren ønsker å se flere aldere 1`] = `
<DocumentFragment>
  <section
    class="_section_1aa5c0"
    data-testid="tidligst-mulig-uttak"
  >
    <p
      class="_ingress_01a375 navds-ingress"
    >
      Du kan tidligst ta ut alderspensjon når du er 62 år og 10 måneder. Hvis du går av senere, får du høyere pensjon i året.
    </p>
  </section>
  <section
    class="_section_1aa5c0 _section_01a375"
  >
    <h2
      class="navds-heading navds-heading--xsmall"
    >
      Når vil du ta ut alderspensjon?
    </h2>
    <ul
      class="navds-chips _chipsWrapper_01a375 _chipsWrapper__hasGap_01a375 navds-chips--medium navds-body-short navds-body-short--small"
    >
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            62 år og 10 md.
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            63 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            64 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            65 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            66 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            67 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            68 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            69 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            70 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            71 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            72 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            73 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            74 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            75 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            76 år
          </span>
        </button>
      </li>
      <li>
        <button
          aria-pressed="false"
          class="navds-chips__chip navds-chips__toggle"
        >
          <span
            class="navds-chips__chip-text"
          >
            77 år
          </span>
        </button>
      </li>
    </ul>
    <button
      class="_visFlereAldere_01a375 navds-button navds-button--tertiary navds-button--xsmall"
    >
      <span
        class="navds-button__icon"
      >
        <svg
          aria-hidden="true"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.47 7.97a.75.75 0 0 1 1.06 0l5.5 5.5a.75.75 0 1 1-1.06 1.06L12 9.56l-4.97 4.97a.75.75 0 0 1-1.06-1.06l5.5-5.5Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        aria-live="polite"
        class="navds-label navds-label--small"
      >
        Vis færre aldere
      </span>
    </button>
  </section>
</DocumentFragment>
`;
